invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalUsageInArrear.java

```diff
@@ -207,9 +214,12 @@ public UsageInArrearItemsAndNextNotificationDate computeMissingItemsAndNextNotif
             // Compare the two and add the missing piece if required.
             if (!billedItems.iterator().hasNext() || billedUsage.compareTo(toBeBilledUsage) < 0) {
                 final BigDecimal amountToBill = toBeBilledUsage.subtract(billedUsage);
+                if (billedUsage.compareTo(BigDecimal.ZERO) > 0){
+                    toBeBilledUsageDetails.add(new ToBeBilledConsumableInArrearDetail(-1, "billedUsage", billedUsage, -1));
```

Comment:
FYI, for the full implementation, I'm not sure we can take such a shortcut unfortunately, as with `DETAIL_MODE`, we might be missing entries for each individual tier.

I believe instead of `computeBilledUsage`, we will need to compare existing vs _proposed_ (to be billed) items for each tier.

Reply:
I am confused about this section: Are we just trying to add some 'details' in the json for the already billed items?

Reply:
@sbrossie, yes it is for the already billed. Is that when I added the detail, the sum is not equal to the total amount of the item, since there could have billed items. Hence the add of a new billed detail on the JSON. But if it is not necessary I will remove it.

Reply:
@wwjbatista Ok, that's makes sense. Keep it the way you have it.